trio/_core/_multierror.py

```diff
@@ -244,7 +357,9 @@ def derive(self, __excs):
         return exc
 
     @classmethod
-    def filter(cls, handler, root_exc):
+    def filter(
+        cls, handler: Callable[[BaseException], BaseException], root_exc: BaseException
```

Comment:
The handler callable can return `None`. Same applies to the other handler callable annotations throughout the module.

Reply:
Fixed in [a3c8963](https://github.com/python-trio/trio/pull/2742/commits/a3c89639020935c534098e7f6cab8a4014e434fe)